src/commands/cmd_context.h

```diff
@@ -41,7 +42,8 @@ CommandCtx *CommandCtx_New
 	ExecutorThread thread,          // Which thread executes this command
 	bool replicated_command,        // Whether this instance was spawned by a replication command.
 	bool compact,                   // Whether this query was issued with the compact flag.
-	long long timeout               // The query timeout, if specified.
+	long long timeout,              // The query timeout, if specified.
+	bool timeout_on_write           // Use timeout on write.
```

Comment:
` // Apply timeout on both read and write queries.`
when `timeout_on_write` is true the timeout is applied to both READ and WRITE queries, in this case maybe we should consider renaming this field to: `rw_timeout` or `timeout_rw` ?